### PR TITLE
OCI GetChart

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

Follow up of #2311

Now it's time to implement the actual interface for the OCI case. 

### Possible drawbacks

As explained in a `TODO`, using a CA cert won't work here. The only alternative with ORAS is to use an `insecure` flag but I didn't included that (yet?).

### Applicable issues

  - ref #2232

### CODE REVIEW

File: pkg/chart/chart.go [hunk 1]
```diff
@@ -373,3 +389,43 @@ func getRegistrySecretsPerDomain(appRepoSecrets []string, cluster, namespace, to
 	}
 	return secretsPerDomain, nil
 }
+
+// InitClient returns an HTTP client based on the chart details loading a
+// custom CA if provided (as a secret)
+// TODO(andresmgot): Using a custom CA cert is not supported by ORAS (neither helm), only using the insecure flag
```
Comment (absoludity) [inline pkg/chart/chart.go:395]: Do you know why (as in, is this intentional to require public tls certs)? What are the implications for private registries?

File: pkg/helm/test/utils.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/kubeapps/kubeapps/pkg/helm"
+)
+
+// CheckHeader verifies that the given puller contains the given header
+func CheckHeader(t *testing.T, puller helm.ChartPuller, key, value string) {
```
Comment (absoludity) [inline pkg/helm/test/utils.go:13]: Nice move :)

**Review by absoludity — APPROVED** [review]
The lack of support for encrypted traffic to a private registry is a bit worrying? Keen to know if this is known, intentional (in oras/helm) or an issue they're working on?

**Comment by andresmgot** [comment]
> The lack of support for encrypted traffic to a private registry is a bit worrying? Keen to know if this is known, intentional (in oras/helm) or an issue they're working on?

This is the main issue for Helm: https://github.com/helm/helm/issues/6324 and the related implementation in ORAS: https://github.com/deislabs/oras/issues/109 but in none of them there are mentions to support a custom CA. In any case, the traffic is encrypted with TLS but the certs are not validated.

**Comment by absoludity** [comment]
> > The lack of support for encrypted traffic to a private registry is a bit worrying? Keen to know if this is known, intentional (in oras/helm) or an issue they're working on?
> 
> This is the main issue for Helm: [helm/helm#6324](https://github.com/helm/helm/issues/6324) and the related implementation in ORAS: [deislabs/oras#109](https://github.com/deislabs/oras/issues/109) but in none of them there are mentions to support a custom CA. In any case, the traffic is encrypted with TLS but the certs are not validated.

I just created https://github.com/deislabs/oras/issues/217 so we can at least be moving toward something not susceptible to machine-in-the-middle.

**Pull request merged**